text/0000-clarity-api-description.md

```diff
@@ -0,0 +1,77 @@
+- Start Date: 2015-06-18
+- RFC PR: (leave this empty)
+- Refract Issue: (leave this empty)
+
+# Summary
+
+Provide clarity and additional elements/properites in API Description namespace.
+
+# Motivation
+
+The purpose of this document is to make it easier for newcomers to understand the format and build tools around it.
+
+# Detailed design
+
+## Combine Namespaces
+
+We should combine the API Description namespace with the resource namespace.
+Right now they are separate and it would be clearer if they were not.
+
+## Elements in Attributes
+
+Refract allows for including elements within the attributes section of an element.
+Even though it’s possible, it may be helpful to avoid this in such an important namespace.
+Reason being, this will be one of the first places people interact with Refract, and to use complex features may be too much.
+
+## HREF Variables
+
+We should call them “hrefVariables” everywhere instead of parameters in places (specifically transitions).
+I think this will be more consistent and easier to understand for newcomers.
+This will make more sense in how we handle inheritance as well, because parameters inheriting from hrefVariables does not seem optimal.
+
+## Defining Classes
+
+Swagger provides a way to define tags that can be applied to resources throughout the document.
+Refract has the “class” attribute that we can use the same way, we only need a way to define a class name and provide the description for it.
+Of course, this won’t carry over to API Blueprint, but that’s OK.
+
+## Clarify Inheritance
+
+We need to clarify how inheritance works by defining the behavior for:
+
+* HREF Variables and HREF values throughout the document
+* Content types in payloads
+* Methods in transitions
+* Resources with schemas (though not in the spec) should be inherited by transitions
+
+## Transitions at Root
+
+We should be able to define transitions on the root of the document and not require they are nested within a resource.
+This type of transition, though, should have a URL and a method.
+
+## Subclass Asset
+
+We should subclass the Asset Element and add one called “messageShema” and another called “messageBody.”
+This will make reading a API Description make more sense than seeing Asset alone with classes.
+
+## Provided Media Types
+
+We should add a section that defines what media types the server will respond with.
```

Comment:
> ...the question is whether this is useful concept for any API (description / adapter) and thus it should be "canonical" or is it specific to Swagger format

Another question: would something like this be valid and useful?

```apib
# My resource [/frobs/{id}{?description}]
+ Attributes
  + id (required)
  + description
## Get a frob [GET]
+ Parameters
  + id (required)
+ Response 200 (application/json)
+ Response 200 (application/yaml)
+ Response 200 (application/xml)
+ Response 200 (text/plain)
## Update a frob [PUT]
+ Parameters
  + id (required)
  + description
+ Request ...
+ Response 200 (application/json)
+ Response 200 (application/yaml)
+ Response 200 (application/xml)
+ Response 200 (text/plain)
...
```

The basic question to ask here I think is whether the request/response examples are expected to be an exhaustive list of all possible returned mimetypes or if that information makes more sense at the root level of the API description document. I think it only makes sense at the root **if the responses can be generated**, e.g. from something like MSON with default values. Otherwise we have a documented & supported response mimetype with no usable example, and this is not specific to API Blueprint or Swagger.